docs/en/06-developer-guide/architecture.md

```diff
@@ -0,0 +1,275 @@
+│   ├── core                              # Software infrastructure
+│   ├── extract_model.py  # Use it to export part of  onnx
+│   ├── inference.py             # Abstract function, which will actually call torch/ncnn specific inference
+│   ├── ncnn                            # ncnn Wrapper
+│   └── visualize.py              # Still an abstract function, which will actually call torch/ncnn specific inference and visualize
+..
+├── backend                  # Backend wrapper
+│   ├── base                            # Because there are multiple backends, there must be an OO design for the base class
+│   ├── ncnn                           # This calls the ncnn python interface for model conversion
+│   │   ├── init_plugins.py           # Find the path of ncnn custom operators and ncnn tools
+│   │   ├── onnx2ncnn.py            # Wrap `mmdeploy_onnx2ncnn` into a python interface
+│   │   ├── quant.py                       # Wrap the `ncnn2int8` tool as a python interface
+│   │   └── wrapper.py                  # Wrap pyncnn forward API
+..
+├── codebase                #  Algorithm rewriter
+│   ├── base                          # There are multiple algorithms here that we need a bit of OO design
+│   ├── mmcls                      #  mmcls related model rewrite
+│   │   ├── deploy                       # mmcls implementation of base abstract task/model/codebase
+│   │   └── models                      # Real model rewrite
+│   │       ├── backbones                 # Rewrites of backbone network parts, such as multiheadattention
+│   │       ├── heads                           # Such as MultiLabelClsHead
+│   │       ├── necks                            # Such as GlobalAveragePooling
+│..
+├── core                         # Software infrastructure of rewrite mechanism
+├── mmcv                     #  Rewrite mmcv
+├── pytorch                 #  Rewrite pytorch operator for ncnn, such as Gemm
+..
+```
+
+Each line above needs to be read, don't skip it.
+
+When typing `tools/deploy.py` to convert ViT, these are 3 things:
+
+1. Rewrite of mmcls ViT forward
+2. ncnn does not support `gather` opr, customize and load it with libncnn.so
+3. Run it with real inference, render output, and make sure the result is correct
+
+### 1. Rewrite `forward`
+
+Because onnx generates too much useless operators, and ncnn does not support onnx perfectly, mmdeploy's solution is to hijack the forward code and change it, the output onnx is suitable for ncnn.
+
+For example, change the process of `conv -> shape -> concat_const -> reshape` to `conv -> reshape` to trim off the redundant `shape` and `concat` operators.
+
+All mmcls algorithm overrides are in the `mmdeploy/codebase/mmcls/models` directory.
+
+### 2. Custom Operator
+
+Operators customized for ncnn are in the `csrc/mmdeploy/backend_ops/ncnn/` directory, and are loaded together with `libncnn.so` after compilation. The essence is in hotfix ncnn, which is currently implemented these operators:
```

Comment:
'which is currently implemented these operators' -> 'which currently implements these operators'

Reply:
done.